src/testlib/include/testlib/MongoTestFixture.hpp

```diff
@@ -29,10 +29,37 @@ class MongoTestFixture {
     static mongocxx::uri connectionUri();
 
 private:
+    class ApmEvent {
+        public:
+        ApmEvent(const std::string& command_name_, const bsoncxx::document::value& document_)
+            : command_name(command_name_), value(document_), command(value.view()) {}
+
+        std::string command_name;
+        bsoncxx::document::value value;
+        bsoncxx::document::view command;
+    };
     mongocxx::instance& instance;
 
 protected:
     mongocxx::client client;
+    std::vector<ApmEvent> events;
+    std::function<void(const mongocxx::events::command_started_event&)> apmCallback =
```

Comment:
```I think forcing callers to manage the lifetime of a `std::vector<ApmEvent>` is easier to deal with than having callers periodically call `clearEvents()` because the former happens automatically when running a `TEST_CASE_METHOD()` with multiple `SECTIONS()`.```

clearEvents() also gets automatically called when running `TEST_CASEMETHOD()` with multiple `SECTIONS()`, and to me it seems like it would be more clear to only need to call clearEvents() since it mimics the dropAllDatabases() paradigm.

The `makeApmCallback` idea would pollute the mongofixture namespace less than the current implementation, is that something we care about?

Reply:
Per my in person discussion with Max, we will want to use the `makeApmCallback` implementation. In the current implementation, if the developer forgets to call `clearEvents()`, the events vector may or may not have previous elements in it, which could inconsistent behavior. With the `makeApmCallback` idea, if the developer forgets to initialize an events vector the test simply won't work, which is a less fragile and error prone approach.